metadata-ingestion/src/datahub/ingestion/run/pipeline.py

```diff
@@ -341,6 +341,9 @@ def _notify_reporters_on_ingestion_start(self) -> None:
                 logger.warning("Reporting failed on start", exc_info=e)
 
     def _notify_reporters_on_ingestion_completion(self) -> None:
+        # This method is being called when there are records to flush
+        # in rest-sink so adding this flush here
+        self.sink.flush()
```

Comment:
I wouldn't assign _flush_ responsibility to the `_notify_reporters_on_ingestion_completion` method but to the one calling it.